posts/superseding-bpmodels/index.qmd

```diff
@@ -0,0 +1,51 @@
+
+One thing is clear from the examples on scope changes - they often lead to name changes. Another important decision was whether to rebrand the package with a new name. A new name can signal a fresh approach and address limitations of the original package. The most popular example is the renaming of `{ggplot}` to [`{ggplot2}`](https://ggplot2.tidyverse.org/index.html). Other examples include renaming {reshape} to {reshape2} and {tidyr}. In our case, we decided to fork the original [`{bpmodels}`](https://epiforecasts.github.io/bpmodels/) repository under Epiverse-TRACE and maintain the old package to avoid breaking scripts that rely on it. At the same time, we introduced {epichains} as the successor. The name reflects the fact that it is a package for analysing *epidemiological transmission chains*.
+
+A second key consideration is the plans that the original package author(s) may have had and their views on any future changes. In our case, this was fairly straightforward because the maintainer of the original package was fully involved in the refactoring. Another package author who had made substantial contributions could be reached and gave their approval. More generally, however, bringing all package authors on board with, for example, changes in scope and name is an important step in taking on maintenance of a package and one that should not be neglected.
+
+We also had to consider whether to archive {bpmodels} or allow it to coexist with {epichains}. We decided to keep {bpmodels} accessible to sustain the reproducibility of existing code using the package. The package was moved back to the [epiforecasts GitHub organisation](https://epiforecasts.github.io/bpmodels/) where it originated from. We, however, added a [lifecycle badge](https://lifecycle.r-lib.org/reference/badge.html) to communicate the package’s retired status and text in the README about our plans for low maintenance.
+
+Another technical consideration was how to handle version control and commit histories. When forking a package, it’s important to decide whether to retain the commit history. Options include squashing the history to start with a clean slate, which risks losing visibility of past contributions, or tagging the HEAD commit of the original repository and building from there. For {bpmodels}, we chose to keep the history intact to retain the contributions of its original authors.
+
+[Semantic versioning](https://semver.org/) was another key decision point. Since `{epichains}` was not going to be available immediately but would be developed in the open (on GitHub), we needed to consider how to communicate that to potential users. We decided to start at version `0.0.0.9999` to signal an experimental and unstable phase[^3] while iterating on features.
```

Comment:
Would it make sense to link to the relevant issues each time so that curious readers can go through the whole discussion?

e.g., https://github.com/epiverse-trace/epichains/issues/32

Reply:
Would it be better as a hyperlink or a footnote? I'm not sure.

Reply:
I'm not sure either. I keep hesitating between both always. In this case, my first intuition would be a link but I don't feel strongly either :shrug: